### PR TITLE
How-To Guide for Using a Property Model

### PR DESCRIPTION
**Pull request by MarcusHolly**
## Fixes/Addresses:

Issue #275 

## Summary/Motivation:

This PR adds a How-To guide for using a property model.

## Changes proposed in this PR:
- Adds an .rst file to the list of How-To guides
- Updates index file

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by aladshaw3** [comment]
Here is the log output from the doc test that failed.
[FailedTest.txt](https://github.com/watertap-org/watertap/files/8564530/FailedTest.txt)
 
Looks like the test doesn't like the fact that you are displaying output...?

File: docs/how_to_guides/how_to_use_a_property_model.rst [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    m.fs.state_block[0].enth_mass_phase['Liq']
+    m.fs.state_block[0].pressure_osm
+
+    # Create the solver object
+    solver = get_solver()
+    solver.options = {'nlp_scaling_method': 'user-scaling'}
+
+    # Solve the model and display the output
+    results = solver.solve(m, tee=True)
+    m.fs.state_block[0].display()
```
Comment (aladshaw3) [inline docs/how_to_guides/how_to_use_a_property_model.rst]: Try setting `tee=False` and remove the line that displays output. See if that gets rid of the failure.

Comment (aladshaw3) [inline docs/how_to_guides/how_to_use_a_property_model.rst]: Try setting `tee=False` and remove the line that displays output. See if that gets rid of the failure. 
```suggestion
    results = solver.solve(m, tee=False)
    #m.fs.state_block[0].display()
```

File: docs/how_to_guides/how_to_use_a_property_model.rst [hunk 2]
```diff
@@ -0,0 +1,70 @@
+How to use a property model
+------------------------------------------------
+
+The example below shows how to use a property model and display outputs for a state block.
+
```
Comment (aladshaw3) [inline docs/how_to_guides/how_to_use_a_property_model.rst]: May also be nice to have some information or language as to why someone may want or need to use a property model in the absence of a unit model.

**Review by aladshaw3 — CHANGES_REQUESTED** [review]
A good start. Need to resolve the failing test (see suggestions above). Also may be nice to have a little more explanation in the beginning.

**Comment by aladshaw3** [comment]
Ok, so now only the test with python 3.8 is failing. 
[NewError_SameTest.txt](https://github.com/watertap-org/watertap/files/8565304/NewError_SameTest.txt)

A warning is being output (see below) that is causing the current error. 

"""
WARNING: The current pynumero_ASL library is version=3, but found version=2.
        Please recompile / update your pynumero_ASL library.
"""

There should be a way in .rst files using `testcode` and `testoutput` where you can just ignore outputs. However, this may be an issue that needs to be resolved in the `setup.py` so that the GitHub actions are installing the most up to date libraries (I am not sure why this error is popping up only here).

Maybe @lbianchi-lbl can help out a bit more here.

**Comment by MarcusHolly** [comment]
> Ok, so now only the test with python 3.8 is failing. [NewError_SameTest.txt](https://github.com/watertap-org/watertap/files/8565304/NewError_SameTest.txt)
> 
> A warning is being output (see below) that is causing the current error.
> 
> """ WARNING: The current pynumero_ASL library is version=3, but found version=2. Please recompile / update your pynumero_ASL library. """
> 
> There should be a way in .rst files using `testcode` and `testoutput` where you can just ignore outputs.

Ok, thanks! I work part-time, so I'll look into this first thing tomorrow morning.

**Comment by adam-a-a** [comment]
> Ok, so now only the test with python 3.8 is failing. [NewError_SameTest.txt](https://github.com/watertap-org/watertap/files/8565304/NewError_SameTest.txt)
> 
> A warning is being output (see below) that is causing the current error.
> 
> """ WARNING: The current pynumero_ASL library is version=3, but found version=2. Please recompile / update your pynumero_ASL library. """
> 
> There should be a way in .rst files using `testcode` and `testoutput` where you can just ignore outputs. However, this may be an issue that needs to be resolved in the `setup.py` so that the GitHub actions are installing the most up to date libraries (I am not sure why this error is popping up only here).
> 
> Maybe @lbianchi-lbl can help out a bit more here.

I think @bknueven might have some ideas as well.

**Comment by bknueven** [comment]
We're still using idaes-ext 2.5.5, but the runners are getting Pyomo 6.4, which contains a newer version of PyNumero.

I think the easiest thing is to resolve #503, going to at least 2.0.0a1 (or https://github.com/IDAES/idaes-pse/commit/2cb6df02923ca775a0235ec0782a5f8133375f6b) and also enforce `pyomo>6.3.0` in setup.py based on https://github.com/Pyomo/pyomo/commit/e6e9b60af075621b12d7a001990cd394e2596e36.

**Review by adam-a-a — DISMISSED** [review]
Looks good to me! Nice job on your first PR, @MarcusHolly! I'm approving this even though we need to resolve the issue with the failed check.

**Comment by adam-a-a** [comment]
@MarcusHolly - just one change I want to suggest (even though I approved already). Can you move `how_to_use_a_property_model` higher up in the index? I think it should be the second entry.

**Review by adam-a-a — DISMISSED** [review]
LGTM

File: docs/how_to_guides/how_to_use_a_property_model.rst [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    m.fs.state_block[0].flow_vol_phase['Liq']
+    m.fs.state_block[0].molality_comp['NaCl']
+    m.fs.state_block[0].visc_d_phase['Liq']
+    m.fs.state_block[0].diffus_phase['Liq']
+    m.fs.state_block[0].enth_mass_phase['Liq']
+    m.fs.state_block[0].pressure_osm
+
+    # Create the solver object.
+    solver = get_solver()
+    solver.options = {'nlp_scaling_method': 'user-scaling'}
```
Comment (bknueven) [inline docs/how_to_guides/how_to_use_a_property_model.rst]:
```suggestion
```
This is already done by default in WaterTAP.

**Review by aladshaw3 — APPROVED** [review]
Looks like all tests are passing now. I am not sure why there was that one python 3.8 failure before, but everything seems good now.

**Review by tarnold17 — APPROVED** [review]

**Pull request merged**